### PR TITLE
Add non-existing check to EXTCODECOPY

### PR DESCRIPTION
**Pull request by eliseygusev**
As of #259  adds non-existence check for opcode EXTCODECOPY.

### CODE REVIEW

**Comment by eliseygusev** [comment]
Hey, @ed255 could you please take a look when you have time? I am not sure who could help review the changes. 
`make test`/`make lint` work locally for me.

File: specs/opcode/3cEXTCODECOPY.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ The `EXTCODECOPY` circuit constrains the values popped from stack, call context/
    - `memory_offset` is at the second position of the stack
    - `code_offset` is at the third position of the stack
    - `size` is at the fourth position of the stack
-   - `code_hash` from the address
+   - `code_hash` from the address if witness value `exists == 1`. Otherwise only lookup the account non-existing proof.
```
Comment (CPerezz) [inline specs/opcode/3cEXTCODECOPY.md:28]: This looks correct to me. We will need to witness/hint the existance of the account even if we do the lookup. But would be nice if someone else could double-check!

**Review by CPerezz — APPROVED** [review]
LGTM. But let's wait for another PR before merging! @icemelon could you assign someone from Scroll?

**Review by silathdiir — APPROVED** [review]
LGTM

**Pull request merged**